crates/l2/sequencer/l1_committer.rs

```diff
@@ -104,58 +90,112 @@ impl Committer {
             validium: committer_config.validium,
         })
     }
+}
 
-    pub async fn run(&mut self) {
-        loop {
-            if let Err(err) = self.main_logic().await {
-                error!("L1 Committer Error: {}", err);
-            }
+// TODO: these are copy pasted from l1_watcher.rs, consider merging them
```

Comment:
@ElFantasma can correct me here, but I think we don't want to merge `L1Watcher`' messages with `L1Committer`'s as they are different actors that answer to different messages.